### PR TITLE
Development

### PR DESCRIPTION
**Pull request by sjelabd**
Removed the tau_threshold from sum_spectra() and added tau_threshold and eup_threshold arguments to Simulation. Also added some code for an eup threshold in print_lines(), but shouldn't be used until the bug with the intensity mismatch is fixed.

### CODE REVIEW

File: molsim/classes.py [hunk 1]
```diff
@@ -1130,6 +1130,8 @@ def __init__(
                     use_obs = False, # flag for line profile simulation to be done with observations
                     add_noise = False, #flag for whether to add noise 
                     noise = None, # Noise level in native units of the simulation to add
+                    tau_threshold = 1000., #Set upper threshold at which to ignore lines for fitting (Samer)
+                    eup_threshold = 0. #Set lower eup threshold [K] at which to exclude transitions (Samer)
```
Comment (laserkelvin) [inline molsim/classes.py:1134]: Does this imply that by default, everything with an upper state energy >= 0. will have `tau` set to zero?

File: molsim/classes.py [hunk 2]
```diff
@@ -1221,6 +1225,11 @@ def _calc_tau(self):
 								self.source.dV*1000 * self.mol.q(self.source.Tex)
 							)
 					)
+		#Set the tau to 0 if it exceeds the tau_threshold so as to to not impact the fit (Samer)
+		self.spectrum.tau[self.spectrum.tau>=self.tau_threshold] = 0.
+		#Set the tau to 0 if it exceeds the eup_threshold (Samer)
+		self.spectrum.tau[self.eup<=self.eup_threshold] = 0.
```
Comment (laserkelvin) [inline molsim/classes.py:1231]: I don't think this is the right sign - should it not be `self.eup >= self.eup_threshold` if you want to set tau to zero if it exceeds the threshold?

**Review by laserkelvin — CHANGES_REQUESTED** [review]

File: molsim/classes.py (hunk 2 above)
Comment (sjelabd) [reply molsim/classes.py:1231]: Generally we want to exclude the lower energy lines, as they will often have effects that we can't account for in simulations. My comment is misleading though

File: molsim/classes.py (hunk 1 above)
Comment (sjelabd) [reply molsim/classes.py:1134]: Everything with an upper state energy greater than 0 will be included. It's meant to be a lower limit

Comment (laserkelvin) [reply molsim/classes.py:1134]: Basically I don't want the default behavior of this function to change - can you guarantee someone else using this function won't have their simulations completely messed up by the default value?

File: molsim/classes.py (hunk 2 above)
Comment (laserkelvin) [reply molsim/classes.py:1231]: Can you rewrite the comment to convey your intention a little bit better then? Just make a local change, commit, push and it should update here.

File: molsim/classes.py (hunk 1 above)
Comment (bmcguir2) [reply molsim/classes.py]: Why not set the defaults for both to None, and add logic to your added code to just skip any changes if these values are None?  That would ensure no weird edge cases you haven't thought of.   I can think of several cases where 1e6 would not be good (not because that's a reasonable value, but because someone might want to test making some lines super optically thick, only to see them disappear).

File: molsim/classes.py (hunk 2 above)
Comment (laserkelvin) [reply molsim/classes.py:1231]: Still not perfect, the comment doesn't describe what the code is doing outside of setting `tau` to zero. Explain a bit what the `eup` part is doing (e.g. what are you identifying to set to zero and _why_)

**Review by laserkelvin — CHANGES_REQUESTED** [review]

File: molsim/classes.py (hunk 1 above)
Comment (sjelabd) [reply molsim/classes.py]: Done!

**Review by laserkelvin — APPROVED** [review]

**Comment by laserkelvin** [comment]
Closing, thanks for your patience @sjelabd 👯‍♂️
